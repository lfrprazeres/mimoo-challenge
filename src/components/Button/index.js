import React from 'react';
import styled from 'styled-components';
import { Button as MuiButton } from '@material-ui/core';
import { device } from '../../utils/deviceSizes';
import { connect } from 'react-redux';
import { useHistory } from 'react-router-dom';

const ButtonStyled = styled.div`
    background-color: ${props => props.bg};
    border-radius: 0;
    height: 50px;
    @media ${device.mobileS} {
        bottom: 0;
        position: absolute;
        width: 100vw;
    };
    @media ${device.laptop} {
        left: 50%;
        transform: translate(-50%, -50%);
        width: 300px;
    }
    button {
        color: ${props => props.color};
        font-size: 17px;
        font-weight: bold;
        height: inherit;
        text-transform: capitalize;
        width: inherit;
    };
`

function Button(props) {
    const history = useHistory();

    function handleClick() {
        if(props.onClick) {
            props.onClick()
        }
        history.push(props.to)
    }

    return (
        <ButtonStyled green={props.colors.green} bg={props.bg} color={props.color}>
            <MuiButton onClick={() => handleClick()}>
                {props.label}
            </MuiButton>
        </ButtonStyled>
    )
};

const mapStateToProps = state => ({
    colors: state.app.defaultColors
});



export default connect(mapStateToProps, null)(Button);