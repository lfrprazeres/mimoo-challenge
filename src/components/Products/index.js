import React, { useState } from 'react';
import { connect } from 'react-redux';
import styled from 'styled-components';
import RoomOutlinedIcon from '@material-ui/icons/RoomOutlined';
import { backgroundChooser } from './utils/backgroundChooser';
import * as animation from '../../utils/animations/viewSlide';
import { TransitionGroup, CSSTransition } from "react-transition-group";

const ProductsStyled = styled.div`
    margin-bottom: 20px;
    ul {
        display: flex;
        margin: 10px 0 0 0;
        padding: 0;
        position: relative;
        &:before {
            background-color: ${props => props.green};
            border-radius: 4px;
            bottom: 0;
            content: "";
            height: 3px;
            left: ${props => props.domSelected === undefined ? "0" : props.domSelected.offsetLeft + "px"};
            position: absolute;
            transition: .2s ease;
            width: ${props => props.domSelected === undefined ? "100px" : (props.domSelected.offsetWidth + 5) + "px"};
        }
        li {
            box-sizing: border-box;
            color: ${props => props.gray};
            cursor: pointer;
            font-size: 18px;
            font-weight: 500;
            list-style-type: none;
            margin-right: 14px;
            padding: 5px;
            text-align: center;
            min-width: 100px;
        }
        .selected {
            color: ${props => props.green};
            font-weight: bold;
            padding-right: 2px;
        }
    }
    .brand {
        margin: 51px 0 -31px -5px;
        .brandTitle {
            color: ${props => props.gray};
            display: flex;
            span {
                font-size: 12px;
                font-weight: bold;
                margin-top: 4px;
            }
        }
        .products {
            display: flex;
            flex-wrap: wrap;
            max-width: 100%;
            width: 87vw;
            .product {
                align-items: flex-end;
                box-shadow: 0px 0px 4px #00000029;
                border-radius: 5px;
                display: flex;
                height: 167px;
                justify-content: center;
                margin: 9px 7px 6px 5px;
                overflow: hidden;
                min-width: 160px;
                .productImage {
                    bottom: -6px;
                    left: -49px;
                    position: relative;
                    img {
                        bottom: -2px;
                        position: absolute;
                        width: 7em;
                    }
                }
        }
        }
        
    }
`;


function Products(props) {
    const [ selected, setSelected ] = useState(0);
    const [ domSelected, setDomSelected ] = useState();
    const {
        products,
        colors
    } = props;
    return (
        <ProductsStyled gray={colors.gray} green={colors.green} domSelected={domSelected}>
            <ul>
                {products.map((item, index) => {
                    return (
                        <li key={index} onClick={(e) => {setSelected(index); setDomSelected(e.target)}} className={selected === index ? "selected" : ""}>
                            {item.category}
                        </li>
                    )
                })}
            </ul>
            {products.length > 0 && 
                <div>
                    {products[selected].brands.slice(0).reverse().map((brand, index) => {
                        return (
                            <div className="brand" key={index}>
                                <div className="brandTitle">
                                    <RoomOutlinedIcon />
                                    <span>
                                        {brand.name}
                                    </span>
                                </div>
                                <div className="products">
                                    {brand.products.map((product,index) => {
                                        let background = backgroundChooser(products[selected].category, colors.beige, colors.aquaBlue);
                                        return (
                                            <div key={index} className="product" style={{background: background}}>
                                                <div className="productImage">
                                                    <img src={product.image} alt={product.name}/>
                                                </div>
                                            </div>
                                        )
                                    })}
                                </div>
                            </div>
                        )
                    })}
                </div>
            }
        </ProductsStyled>
    )
};

const mapStateToProps = state => ({
    products: state.user.products,
    colors: state.app.defaultColors

});

export default connect(mapStateToProps, null)(React.memo(Products));