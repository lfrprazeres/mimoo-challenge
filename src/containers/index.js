import SplashScreen from './SplashScreen';
import Nome from './Nome';
import Home from './Home';
import Scan from './Scan';
import Confirmacao from './Confirmacao';

export {
    SplashScreen,
    Nome,
    Home,
    Scan,
    Confirmacao
}