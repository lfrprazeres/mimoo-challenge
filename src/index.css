@import url('https://fonts.googleapis.com/css2?family=Poppins:wght@100;500;600;700&display=swap');
@import url('https://fonts.googleapis.com/css2?family=Montserrat:wght@100;400;500;600;700&display=swap');

* {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
  font-family: 'Poppins', sans-serif;
}

html,body {
  min-width: 100%;
  min-height: 100%;
  overflow-x: hidden;
}

a {
  text-decoration: none;
}

li {
  list-style-type: none;
}