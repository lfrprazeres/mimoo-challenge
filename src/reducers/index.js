import { combineReducers } from 'redux';
import { appReducer } from './app';
import { userReducer } from './user';

export const Reducers = combineReducers({
    app: appReducer,
    user: userReducer
})