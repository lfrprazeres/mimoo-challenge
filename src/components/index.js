import Button from './Button';
import Products from './Products';

export {
    Button,
    Products
}