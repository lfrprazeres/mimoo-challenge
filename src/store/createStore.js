
import { createStore, applyMiddleware } from 'redux';
import createSagaMiddleware from 'redux-saga'
import Saga from '../saga';
import { Reducers } from '../reducers';
import { composeWithDevTools } from 'redux-devtools-extension';
import { apiMiddleware } from 'redux-api-middleware';
import { persistStore, persistReducer } from 'redux-persist';
import storage from 'redux-persist/lib/storage';

const persistConfig = {
    key: 'root',
    storage,
}

const persistedReducer = persistReducer(persistConfig, Reducers);

const sagaMiddleware = createSagaMiddleware();

let store = createStore(
    persistedReducer,
    composeWithDevTools(
        applyMiddleware(
            sagaMiddleware,
            apiMiddleware
        )
    )
);

sagaMiddleware.run(Saga);

const persistor = persistStore(store);
export { store, persistor }