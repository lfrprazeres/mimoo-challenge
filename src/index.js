import React from 'react';
import ReactDOM from 'react-dom';
import './index.css';
import Routes from './routes';
import { Provider } from 'react-redux';
import { store, persistor } from './store/createStore';
import { PersistGate } from 'redux-persist/integration/react';

ReactDOM.render(
  <React.StrictMode>
    <Provider store={store}>
      <PersistGate loading={null} persistor={persistor}>
        <Routes />
      </PersistGate>
    </Provider>
  </React.StrictMode>,
  document.getElementById('root')
);
