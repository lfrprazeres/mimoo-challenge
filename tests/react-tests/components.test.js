import React from 'react';

import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { Provider } from "react-redux";
import { store } from '../../src/store/createStore';
import { Button, Products } from '../../src/components';

configure({adapter: new Adapter()});

describe('Components tests', () => {

    it('should render Button without crash with the properly props', () => {
        const props = {
            to: "/",
            label: "test",
            bg: "white",
            color: "green",
            onClick: jest.fn()
        }
        const wrapper = shallow(
            <Provider store={store}>
                <Button {...props}/>
            </Provider>
        )
        expect(wrapper).toMatchSnapshot();
    })
    it('should render Products without crash with the properly props', () => {
        const wrapper = shallow(
            <Provider store={store}>
                <Products />
            </Provider>
        )
        expect(wrapper).toMatchSnapshot();
    });

    
})