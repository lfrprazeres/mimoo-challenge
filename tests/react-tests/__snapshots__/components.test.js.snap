// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components tests should render Button without crash with the properly props 1`] = `ShallowWrapper {}`;

exports[`Components tests should render Products without crash with the properly props 1`] = `ShallowWrapper {}`;
